ibm/service/power/data_source_ibm_pi_sap_profiles.go

```diff
@@ -106,7 +118,18 @@ func dataSourceIBMPISAPProfilesRead(ctx context.Context, d *schema.ResourceData,
 	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
 
 	client := instance.NewIBMPISAPInstanceClient(ctx, sess, cloudInstanceID)
-	sapProfiles, err := client.GetAllSAPProfiles(cloudInstanceID)
+	filters := map[string]string{
+		Arg_FamilyFilter: "",
+		Arg_PrefixFilter: "",
+	}
+	if v, ok := d.GetOk(Arg_FamilyFilter); ok {
+		filters[Arg_FamilyFilter] = v.(string)
+	}
+	if v, ok := d.GetOk(Arg_PrefixFilter); ok {
+		filters[Arg_PrefixFilter] = v.(string)
+	}
+
+	sapProfiles, err := client.GetAllSAPProfilesWithFilters(cloudInstanceID, filters[Arg_FamilyFilter], filters[Arg_PrefixFilter])
```

Comment:
I'm thinking it would be better for the SDK function to take in the map instead of the individual arguments in case later down the line there are additional filters. I'm not sure how likely that is however.